src/app.ts

```diff
@@ -270,17 +276,29 @@ export class App {
    * @example
    * // Decorate context with getter
    * app.decorateContext('helloWorld', {get: () => 'Hello World!'});
+   *
+   * // Decorate context with nested getter
+   * app.decorateContext('hello.World', {get: () => 'Hello World!'});
    */
   decorateContext(name: string, fn: Decoration): this {
     const proto: MojoContext = Context.prototype as MojoContext;
-    if (Object.getOwnPropertyDescriptor(proto, name) != null) {
-      throw new Error(`The name "${name}" is already used in the prototype chain`);
+    const nestedNames = name.split('.');
+    const baseName = nestedNames[0];
+    const methodName = nestedNames.pop() ?? '';
+    const target = nestedNames.reduce((obj: any, key) => (obj[key] = obj[key] || {}), this._contextClass.prototype);
+
+    if (Object.getOwnPropertyDescriptor(proto, baseName) != null) {
+      throw new Error(`The name "${baseName}" is already used in the prototype chain`);
     }
 
     if (typeof fn !== 'function') {
-      Object.defineProperty(this._contextClass.prototype, name, fn);
+      Object.defineProperty(
+        target,
+        methodName,
+        nestedNames.length == 0 ? fn : {get: fn.get?.bind(this), set: fn.set?.bind(this)}
+      );
     } else {
-      this._contextClass.prototype[name] = fn;
+      target[methodName] = nestedNames.length == 0 ? fn : fn.bind(this);
```

Comment:
Doesn't this bind to the app instance? And the helpers/decorators in the tests only work by accident because `app.config` exists too?

Reply:
A test using `ctx.stash` might be safer to ensure it really works.

Reply:
- Yes, it binds to the app instance, other properties like ctx.stash cannot be accessed.
- I added a test using ctx.stash (it fails so far)